rfc/rfc000-package-server.md

```diff
@@ -1,19 +1,91 @@
+# Package Server
+
 ---
+
 RFC: 000
-Author: Peter Bourgon <peter@bourgon.org>
+
+Author: Matt Heath <matt@mattheath.com>
+
 Status: Draft
----
 
-# package server
+---
 
 ## Motivation
 
-http://peter.bourgon.org/go-kit/#package-server
+[http://peter.bourgon.org/go-kit/#package-server](http://peter.bourgon.org/go-kit/#package-server)
+
+Package server is probably the biggest and most important component of the toolkit. Ideally, we should be able to write our services as implementations of normal, nominal Go interfaces, and delegate integration with the environment to the server package. The package should encode and enforce conventions for server-side concerns, like health checks, system-wide request tracing, connection management, backpressure and throttling, and so on. For each of those topics, it should provide interfaces for different, pluggable strategies. It should integrate with service discovery, and work equally well over multiple transports. Considerable prior art exists in the form of Finagle, Karyon (Netflixs application service library), and likely many more.
 
 ## Scope
 
-To be defined.
+### Endpoints
+
+*   An endpoint is defined as a handler interface which receives a request and returns a response or an error to the client.
```

Comment:
I certainly agree that there needs to be a `Context` somewhere in here. I also agree with @dsymonds that it ought to be its own parameter... Ideally the `Request` type will be pure structural payload data that can be serialized and deserialized without losing information, and I wouldn't want the Context to be part of that serialized format. (I'm imagining that `Request` could (or even should) be a protobuf or thrift type here)

I might be misunderstanding what you mean by `Request` here, though... is that supposed to be a user-defined type? If so, clearer to call it `MyRequest`. If it is in fact a general-purpose Request wrapper that has the app-specific payload data along with other tidbits, then perhaps the Context could go in there, too. Though I would prefer to see that parameter represent the user payload directly if possible.